### PR TITLE
Propose rework of chain-api and chain-connect type handling

### PR DESCRIPTION
**Pull request by sentientforest**
Removes need for duplication / redeclaration of chain-api classes for consuming libraries.
Standardize type transformation / validation across sub-projects of the mono repo
Use stronger / static typing

### CODE REVIEW

**Review by befitsandpiper — APPROVED** [review]
Looks good, I like the types simplification, thanks!

File: chain-connect/src/chainApis/TokenApi.ts [hunk 1]
```diff
@@ -13,45 +13,46 @@
...
-  RefreshAllowanceParams,
-  ReleaseTokenParams,
+  MintTokenDto,
+  MintTokenWithAllowanceDto,
+  RefreshAllowanceDto,
+  ReleaseTokenDto,
   TokenAllowanceBody,
+  TokenBalance,
   TokenBalanceBody,
   TokenBurnBody,
```
Comment (erikgala) [inline chain-connect/src/chainApis/TokenApi.ts:46]: Are the body types still needed? I think those were also constructorArg wrapped types?

**Comment by IndiaJonathan** [comment]
I think ideally we don't remove the types, allowing those makes devlopment much more intuitive. A typical dev is going to want to define the DTO with the type rather than always having to instantiate a class IMO

It also keeps it from being async

**Comment by IndiaJonathan** [comment]
Basically, I think it's really important to keep the ability to do things like this:

```
  arbitraryToken: TokenClassKeyBody = {
    additionalKey: 'fakeTokenKey',
    category: 'fakeCategory',
    collection: 'fakeCollection',
    type: 'fakeType'
  }

  arbitraryTokenInstance: TokenInstanceKeyBody = { ...this.arbitraryToken, instance: '0' }
  
```

**Pull request merged**